### PR TITLE
Update AsyncAPI component to 1.0.0-x releases

### PR DESCRIPTION
**Pull request by kaiszybiak**
## Hey, I just made a Pull Request!

We are using [asyncapi/asyncapi-react](https://github.com/asyncapi/asyncapi-react) in Backstage to display AsyncAPI definitions. This package is still <1.0.0 and got a new version on the road to 1.0.0. This version is a complete rewrite, if we want to update to the newer version, we have to make sure that it still works as before.

**Light Theme:**
142037785-2cf95270-a2d9-41cf-a5d5-1a28f6b7a5e8

**Dark Theme:**
142037807-9b300007-5e15-46b4-8e35-5b7636c6cbd1
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — APPROVED** [review]
Approved from a code standpoint.

My ability to test this thoroughly feels a bit limited, but the screenshots look good. I presume you did try this out before and after the change and verified that the results seemed to be the same and made sense?

**Comment by kaiszybiak** [comment]
After some more investigation and testing we noticed a problem with how we integrate the `@asyncapi/react-component/styles/default.css`. Right now the component specific style is globally available but it should be scoped to the context of the component. For semantic CSS classes this might not be a problem but in our case it affects the button focus of the whole application and other components might be affected as well. 

````css
/* snippet from @asyncapi/react-component/styles/default.css */

button:focus {
  outline: 1px dotted;
  outline: 5px auto -webkit-focus-ring-color;
}
````

We had different approaches how to handle this obstacle and would like to collect your feedback before continuing our thought and decision process:

1. Try to merge this [PR](https://github.com/asyncapi/asyncapi-react/pull/381) in the [asyncapi-react](https://github.com/asyncapi/asyncapi-react) repo, which introduces prefixes for css classes. --> This might take a while + it was automatically marked as stale because it has not had recent activity

2. Create a global solution to scope imported CSS to the component context. 
We noticed that there are already multiple CSS imports in other plugins which are loaded globally.
Example: (`plugins/graphiql/src/components/GraphiQLPage/GraphiQLPage.tsx` --> `import 'graphiql/graphiql.css';`)

3. Configure `css-loader` to support [CSS Modules](https://github.com/webpack-contrib/css-loader#modules) / [CSSStyleSheet](https://github.com/webpack-contrib/css-loader#exporttype) and integrate external stylesheets into `makeStyles()`

Right now our favourite option would be 1. but for the long term we should think about a global solution to avoid such conflicting styles in general. 

You might have already thought about this problem? We would be happy to hear about other possible solutions 🙂 @freben

**Comment by 4integration** [comment]
Related? https://github.com/backstage/backstage/issues/8228

**Review by Rugvip — COMMENTED** [review]
@kaiszybiak Thank you for the thorough description on the style issue! I think (1) is probably the simplest path for now, unless there's someone from the community that wants to look into build system support for scoping stylesheets. For example the selectors in `graphiql.css` are well namespaces, and to be honest I would generally expect that libraries that export stylesheets to be loaded directly should always have those styles be well namespaced.

Bit of a warning on the `makeStyles` thread is that MUI 5 is moving away from that style system so we should likely avoid adding solutions using that method.

File: plugins/api-docs/package.json
Comment (kaiszybiak) [reply plugins/api-docs/package.json]: Hey @Rugvip thanks for your feedback 🙂 According to [this](https://github.com/asyncapi/asyncapi-react/pull/493#issuecomment-986639603) comment I would assume that the release for 1.0 should happen soon. We just merged [this](https://github.com/asyncapi/asyncapi-react/pull/493) fix and I updated this PR to include it so now there are no global styles affecting other components, as everything is scoped within AsyncAPI. From my perspective it would be ok to go with this pre-release version. I would prefer to do another version bump instead of having this PR on hold 🙂.

Comment (Rugvip) [reply plugins/api-docs/package.json]: Alright, thank you! 👍 

If you'd prefer that way I'd say we're good to :shipit: 😁

**Review by Rugvip — APPROVED** [review]

File: plugins/api-docs/package.json
Comment (kaiszybiak) [reply plugins/api-docs/package.json]: awesome 🙂 As I don't have write access could you merge the PR @Rugvip or is there anything I can do from my side ? 🙂

**Comment by dschwank** [comment]
We just noticed that the current AsyncAPI version throws some `ResizeObserver loop limit exceeded` if the container width is smaller than `1280px`.

That error can be ignored safely - but it is still not nice as e.g. Cypress will fail. We opened an issue (see https://github.com/asyncapi/asyncapi-react/issues/498) and will try to support to get it solved.

Will keep you updated. 👋

**Comment by magicmatatjahu** [comment]
FYI fixed in the `1.0.0-next.26` version :)

**Pull request merged**
